### ISSUE TITLE
Add workflow_dispatch trigger for GitHub Actions jobs

### ISSUE THREAD

**Issue by 54a66982-c128-4a20-bcce-ef4a26c3f3a8** [python/cpython#89135]
BPO | [44972](https://bugs.python.org/issue44972)
--- | :---
Nosy | @brettcannon, @vstinner, @ambv, @zware, @zooba, @pablogsal, @erlend-aasland, @FFY00, @nightlark
PRs | <li>python/cpython#27873</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-08-21.17:23:48.452>
labels = ['build', '3.8', '3.9', '3.10', '3.11', 'type-feature', '3.7']
title = 'Add workflow_dispatch trigger for GitHub Actions jobs'
updated_at = <Date 2021-08-27.05:18:37.834>
user = 'https://github.com/nightlark'
```

bugs.python.org fields:
```python
activity = <Date 2021-08-27.05:18:37.834>
actor = 'erlendaasland'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Build']
creation = <Date 2021-08-21.17:23:48.452>
creator = 'rmast'
dependencies = []
files = []
hgrepos = []
issue_num = 44972
keywords = ['patch']
message_count = 10.0
messages = ['400036', '400038', '400145', '400146', '400147', '400149', '400157', '400165', '400397', '400398']
nosy_count = 9.0
nosy_names = ['brett.cannon', 'vstinner', 'lukasz.langa', 'zach.ware', 'steve.dower', 'pablogsal', 'erlendaasland', 'FFY00', 'rmast']
pr_nums = ['27873']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue44972'
versions = ['Python 3.7', 'Python 3.8', 'Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by erlend-aasland** [comment]
@ambv should this be backported, or can we close this?

**Comment by ambv** [comment]
3.9 and older are now in security-only fixes so I'd leave them alone. We could do a backport to 3.10 if you're up for it.

**Comment by kumaraditya303** [comment]
I closed this as no backport for 3.10 was created, if anyone wants to do that then feel free to reopen.
